clients/drcachesim/tools/view.cpp

```diff
@@ -224,37 +248,17 @@ view_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
             return true; // Do not count toward -sim_refs yet b/c we don't have tid.
         case TRACE_MARKER_TYPE_FILETYPE:
             // We delay printing until we know the tid.
-            if (filetype_ == -1) {
-                filetype_ = static_cast<intptr_t>(memref.marker.marker_value);
-            } else if (filetype_ != static_cast<intptr_t>(memref.marker.marker_value)) {
+            if (filetype_ == OFFLINE_FILE_TYPE_DEFAULT) {
```

Comment:
See comment above about suddenly treating this as a sentinel.

Reply:
Reverted. See https://github.com/DynamoRIO/dynamorio/issues/7228.

Reply:
Was not aware of that: but it makes more sense in opcode_mix which *must* decode and so it can't take in a trace without an arch bit set.  OTOH, the view tool supports traces without encodings, so it does *not* need an arch bit set.